### PR TITLE
[WEB-4025] Autogenerate icon manifests

### PR DESCRIPTION
**Pull request by jamiehenson**
## Jira Ticket Link / Motivation

No specific motivation, the motivation here is to get rid of the manual (and prone to human error) process of adding new icons.

## Summary of changes

Currently, when adding/changing icons, we have to manually include it in an array of icon names that matches its function in `Icon/types.ts`, and preferably remember to reorder that list alphabetically. This is tedious and prone to human error.

This PR extends the existing spritemap generation script (also now ported to TS since we have `ts-node` as a dependency from a previous PR) to generate these arrays automatically based upon icons in the directory, which means that all icons in the package are represented and available, visible in Storybook, and typed.

## How do you manually test this?

Pull down, run Storybook, watch for 
```
🖼️  SVG sprites and icon manifest generated successfully!
```
in the terminal, make sure all the icons are there on the Icons story.

## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

## Release Notes

- **New Features**
  - Introduced a new script for generating SVG sprites, enhancing icon management.
  - Updated the build process to include sprite generation.

- **Improvements**
  - Streamlined icon management by consolidating icon definitions into a centralized structure.

- **Bug Fixes**
  - Corrected icon references in the multiplayer collaboration and reference guide sections for accurate display.

- **Chores**
  - Updated project dependencies and versioning in `package.json`.
  - Modified the `.gitignore` file to include new patterns for ignored files.

### CODE REVIEW

File: scripts/generate-sprites.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  svg: {
+    transform: [
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      function (svg: any) {
+        let globalDefs = "";
+
+        return svg
+          .replace(/<defs>(.+?)<\/defs>/g, (_match: string, def: string) => {
+            globalDefs += def;
+          })
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply scripts/generate-sprites.ts:28]: Not applicable

File: scripts/generate-sprites.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      const iconJsonPath = path.resolve(
+        __dirname,
+        "../src/core/Icon/computed-icons.ts",
+      );
+
+      const generatedIconGroups =
+        `// AUTOGENERATED BY build:sprites - DO NOT EDIT\n\nexport const computedIcons = ${JSON.stringify(iconGroups, null, 2)}`.replace(
+          /]/g,
+          "] as const",
+        );
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply scripts/generate-sprites.ts:91]: Nested arrays will not be present, we wish to add `as const` after each array in this object as written.

File: scripts/generate-sprites.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
+import path from "path";
```
Comment (jamiehenson) [inline scripts/generate-sprites.ts:1]: Only L42, 48-59, 85-97 are new here really

**Review by aralovelace — APPROVED** [review]
This is nice, So I dont need to add the iccon name in the component.

**Comment by jamiehenson** [comment]
@aralovelace that's right yep, just need to add it to the folder with a consistent file name (it's grouped by the bit after icon, i.e. `icon-_display_-fluffy-cat`

**Pull request merged**